### ISSUE TITLE
Pty host output channels are localized inconsistently

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#192787]
Maybe "pty host" should not be localized at all, only the remote part

![Image](https://github.com/microsoft/vscode/assets/2193314/e8e85013-e511-435a-8183-1b51a830d7c2)

**Comment by Tyriar** [comment]
@TylerLeonhardt actually, this is probably because the channel name text is not localized on the remote side/bundle. Is this a known issue?

**Comment by TylerLeonhardt** [comment]
what are "other processes" here?

**Comment by Tyriar** [comment]
The pty host channel is c created on the pty host process and looks are forwarded back to the renderer.

You can see network requests is also not localized only for the remote car which would probably be on the server process?

**Comment by TylerLeonhardt** [comment]
This is working for the extension host:
<img width="184" alt="image" src="https://github.com/microsoft/vscode/assets/2644648/18c09ab0-4256-4ebe-af90-fb3c99e6ea42">

we pass the language that the client is configured to here:
https://github.com/microsoft/vscode/blob/5a400e53e985dc5e24f6ee574b07ab23943841c5/src/vs/workbench/services/extensions/common/remoteExtensionHost.ts#L105

Do you do something similar in the pty host?

**Comment by Tyriar** [comment]
@TylerLeonhardt the extension host is handled specially:

https://github.com/microsoft/vscode/blob/406fab60ac8f62b9032afc79fb15c5122ebd19f9/src/vs/workbench/api/common/extHostLogService.ts#L21

**Comment by TylerLeonhardt** [comment]
Yeah that's just marking the string as a localized string, but if the language of the process is still just "English" then of course the English strings will show up.

This link shows how we set the language of the extension host:
https://github.com/microsoft/vscode/blob/5a400e53e985dc5e24f6ee574b07ab23943841c5/src/vs/workbench/services/extensions/common/remoteExtensionHost.ts#L105

which eventually sets this environment variable:
https://github.com/microsoft/vscode/blob/ba3978f8658ec5048d7d09c41016747ac997358b/src/vs/server/node/extensionHostConnection.ts#L46

and this is how the language of the process is set.

Is the pty host service doing this?

**Comment by Tyriar** [comment]
It doesn't yet, thanks 👍 

https://github.com/microsoft/vscode/blob/406fab60ac8f62b9032afc79fb15c5122ebd19f9/src/vs/server/node/extensionHostConnection.ts#L27

https://github.com/microsoft/vscode/blob/406fab60ac8f62b9032afc79fb15c5122ebd19f9/src/vs/platform/terminal/node/nodePtyHostStarter.ts#L26-L33

**Comment by TylerLeonhardt** [comment]
There may be other processes that don't do this... 

* Network Requests
* Extension Host (webworker)

seem to not.

<img width="209" alt="image" src="https://github.com/microsoft/vscode/assets/2644648/6389f83f-fbe3-4faf-b397-b4c353d16ea7">

Do you know how Network Requests work? Is that a separate process?

**Comment by TylerLeonhardt** [comment]
The webworker one is weird... because there isn't a concept of env vars so `VSCODE_NLS_CONFIG` isn't automatically defined there. I'm not seeing us configure the AMD loader with the language which is why that one isn't working. Let me get a separate issue for that.
